### DISCUSSION TITLE
SessionId or ChatId for multiple users

### DISCUSSION THREAD

**Discussion by LeandroTecware** [FlowiseAI/Flowise#1020] [Q&A] [answered]
I'm concerned about using the system in production with multiple users. In my first tests I noticed that the memory is not clean and continues to pass a large number of touches for the next request. 

I'm using it through API communication and I don't know how to differentiate one user from another. Is there any way to send some SessionId or ChatId identification to unique users and conversations?

**Comment by HenryHengZJ** [comment, marked as answer]
for that you'll have to use 3rd party memory node like Zep, Redis, DynamoDB and pass in a `sessionId`, the normal BufferMemory wont work for multiple users. https://docs.flowiseai.com/memory/zep-memory
